mule-module-apikit/src/main/java/org/mule/module/apikit/UrlUtils.java

```diff
@@ -89,9 +91,20 @@ public static String getBasePath(MuleMessage message)
         if (path == null)
         {
             path = message.getInboundProperty("http.listener.path");
-            if (path != null && path.endsWith("/*"))
-            {
-                path = path.substring(0, path.length() - 2);
+
+            if (path != null) {
+                if (path.endsWith("/*"))
+                {
+                    path = path.substring(0, path.length() - 2);
+                }
+
+                ParameterMap uriParams = message.getInboundProperty("http.uri.params");
+                if (!uriParams.isEmpty()) {
+                    for (Entry<String, String> entry : uriParams.entrySet()) {
+                        String uriParameter = "{" + entry.getKey() + "}";
+                        if (path.contains(uriParameter)) path = path.replace(uriParameter, entry.getValue());
+                    }
+                }
```

Comment:
I think this block should be inside the `if (path.endsWith("/*"))` block

Reply:
If path doesn't end with "/*" the request won't pass the HttpListener and won't make it to the router